src/store/wallet.js

```diff
@@ -36,45 +42,48 @@ function _useWallet(
   const [networkSeed, setNetworkSeed] = useState(Maybe.Nothing());
   const [networkRevision, setNetworkRevision] = useState(Maybe.Nothing());
 
-  const setWalletType = walletType => {
+  const setWalletType = useCallback(walletType => {
```

Comment:
Can you tldr me on what this achieves?

Reply:
useCallback memoizes a function so that it has referential equality between renders where the dependencies don't change.

pre-`useCallback`, any time a component `_useWallet`ed, it would get a new value for `setWalletType`. If it passed that `setWalletType` to a component as props, that component would re-render every time, since `setWalletType` is a unique reference every render. With useCallback, only when the dependencies of `setWalletType` (which is only the `_setWalletType` function) change does `setWalletType` become a new function, meaning we can avoid a re-render in any child we pass this function to.

In this case, `_setWalletType` is the only dependency, and the state setting functions from `useState` have stable identities between renders, so it's safe to omit.

> Even in that case, dispatch identity is still guaranteed to be stable between re-renders. So you may omit it from the effect deps if you want. It’s not going to cause the effect to re-run.

(from dan's blog post below)

I will include the dependencies anyway, though, since being explicit is a great idea.

would recommend https://overreacted.io/a-complete-guide-to-useeffect/ and https://kentcdodds.com/blog/usememo-and-usecallback for reference